src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -483,8 +483,15 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
 
     fun addFun(funSpec: FunSpec) = apply {
       if (kind == Kind.INTERFACE) {
-        requireExactlyOneOf(funSpec.modifiers, KModifier.ABSTRACT)
-        requireExactlyOneOf(funSpec.modifiers, KModifier.PUBLIC, KModifier.PRIVATE)
+
+        if(funSpec.body.isEmpty()) {
+          require(!funSpec.modifiers.contains(KModifier.PRIVATE), {"Abstract $kind function can not be private"})
```

Comment:
nit: it would be more consistent to have lambda outside the function parentheses:

```kotlin
require(!funSpec.modifiers.contains(KModifier.PRIVATE)) {
  "Abstract $kind function can not be private"
}
```